backend/internal/applygrpcsvc/svc.go

```diff
@@ -33,25 +35,25 @@ func (s *server) Apply(ctx context.Context, req *connect.Request[applyv1.ApplyRe
 		return nil, connect.NewError(connect.CodeInvalidArgument, err)
 	}
 
-	logs, err := s.apply.Apply(ctx, msg.Manifest, msg.Path)
+	man, err := sdkmanifest.Read([]byte(msg.Manifest), msg.Path)
 	if err != nil {
 		return nil, connect.NewError(connect.CodeInvalidArgument, err)
 	}
 
-	return connect.NewResponse(&applyv1.ApplyResponse{Logs: logs}), nil
-}
+	var logs []string
 
-func (s *server) Plan(ctx context.Context, req *connect.Request[applyv1.PlanRequest]) (*connect.Response[applyv1.PlanResponse], error) {
-	msg := req.Msg
-
-	if err := proto.Validate(msg); err != nil {
-		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	actions, err := sdkmanifest.Plan(ctx, man, s.client, sdkmanifest.WithLogger(func(msg string) {
+		logs = append(logs, fmt.Sprintf("[plan] %s", msg))
+	}))
+	if err != nil {
+		return nil, connect.NewError(connect.CodeUnknown, err)
 	}
 
-	logs, err := s.apply.Plan(ctx, msg.Manifest)
-	if err != nil {
-		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	if err = sdkmanifest.Execute(ctx, actions, s.client, func(msg string) {
```

Comment:
having plan and execute separate is a good practice, not necessarily in order to not run separately but definitely for testing. also this gives the user a chance to go over the plan and maybe even prune/change it if needed. locking resources is complicated, not sure we want to do this?

Reply:
Maybe not locking between planning and applying, after all there's no guarantee that the user will run "apply" at all, but **during** applying it's a good idea to prevent a potential race between 2 users - but again, we don't even support multiple users right now.

Reply:
Maybe add a "locked" property to projects that only AK can modify? (while it's creating/updating a project)

Reply:
working with terraform, if you plan and save the plan result, 
you can apply the plan and are guaranteed that the end result is what you have in plan 
so in case something changed the state between plan and apply (which can be zero time, days or never), what you planned is what you get